hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -375,6 +376,8 @@ public void run() {
                 } else {
                     handleAuthenticationFailure(endpoint, request);
                 }
+            } catch (QueryResultSizeExceededException e) {
```

Comment:
I'm not 100% sure about this part. I'm just leaving out to log the exception here, which I think should be fine. We don't log this exception at any other proxy. It is meant for the customer who called the method, like a back pressure exception. We should not put a whole stack trace in the log file IMHO.